### ISSUE TITLE
metadata.resourceVersion: Invalid value: \"\"

### ISSUE THREAD

**Issue by jicki** [VictoriaMetrics/operator#1200]
`{"level":"info","ts":"2024-12-21T06:34:46Z","logger":"controller.VMAlert","msg":"updating service vmalert-victoria-metrics-cluster configuration, is_current_equal=false, is_prev_equal=true, is_prev_nil=false","vmalert":"victoria-metrics-cluster","namespace":"monitor-system"}
{"level":"error","ts":"2024-12-21T06:34:46Z","logger":"manager","msg":"Reconciler error","controller":"vmalert","controllerGroup":"operator.victoriametrics.com","controllerKind":"VMAlert","VMAlert":{"name":"victoria-metrics-cluster","namespace":"monitor-system"},"namespace":"monitor-system","name":"victoria-metrics-cluster","reconcileID":"112e8f65-2bca-46ea-80cc-6ea1071628b5","error":"cannot reconcile service for vmalert: Service \"vmalert-victoria-metrics-cluster\" is invalid: metadata.resourceVersion: Invalid value: \"\": must be specified for an update"}
`

`{"level":"error","ts":"2024-12-21T06:33:59Z","logger":"manager","msg":"Reconciler error","controller":"vmagent","controllerGroup":"operator.victoriametrics.com","controllerKind":"VMAgent","VMAgent":{"name":"victoria-metrics-cluster","namespace":"monitor-system"},"namespace":"monitor-system","name":"victoria-metrics-cluster","reconcileID":"64b04744-1271-4826-b6e0-cdea676a8a8e","error":"cannot reconcile service for vmagent: Service \"vmagent-victoria-metrics-cluster\" is invalid: metadata.resourceVersion: Invalid value: \"\": must be specified for an update"}
`

get  service

```
kubectl -n monitor-system get service vmagent-victoria-metrics-cluster -o yaml
apiVersion: v1
kind: Service
metadata:
  annotations:
    meta.helm.sh/release-name: vm-cluster
    meta.helm.sh/release-namespace: monitor-system
  creationTimestamp: "2024-12-21T06:19:38Z"
  finalizers:
  - apps.victoriametrics.com/finalizer
  labels:
    app.kubernetes.io/component: monitoring
    app.kubernetes.io/instance: victoria-metrics-cluster
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: vmagent
    managed-by: vm-operator
  name: vmagent-victoria-metrics-cluster
  namespace: monitor-system
  ownerReferences:
  - apiVersion: operator.victoriametrics.com/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: VMAgent
    name: victoria-metrics-cluster
    uid: a3538539-24f1-4b4f-bd62-8091f0ea3492
  resourceVersion: "1456119786"
  selfLink: /api/v1/namespaces/monitor-system/services/vmagent-victoria-metrics-cluster
  uid: a1ad4de8-6f07-43ae-87b5-e876be60b04f
```

**Comment by f41gh7** [comment]
Hello, could you please share your kubernetes cluster version?

**Comment by jicki** [comment]
> Hello, could you please share your kubernetes cluster version?

kubernetes clusyer version  v1.17.6

**Comment by f41gh7** [comment]
> > Hello, could you please share your kubernetes cluster version?
> 
> kubernetes cluster version v1.17.6

 Thanks! An Issue seems to be related to the old Kubernetes version, before v1.23 version API server requires `metadata.resourceVersion` to be set.

 This issue will be fixed at upcoming `v0.51.2` bugfix release

**Comment by f41gh7** [comment]
The issue was fixed at [v0.51.2](https://github.com/VictoriaMetrics/operator/releases/tag/v0.51.2) release
